include/deal.II/base/numbers.h

```diff
@@ -729,8 +727,8 @@ namespace internal
   template <typename T>
   struct NumberType
   {
-    static constexpr DEAL_II_ALWAYS_INLINE DEAL_II_CUDA_HOST_DEV const T &
-    value(const T &t)
+    static constexpr KOKKOS_FORCEINLINE_FUNCTION const T &
+                                                       value(const T &t)
```

Comment:
To give a more illustrative example of what @peterrum wrote above, I find this usage annoying: We have several dozen of `DEAL_II_ALWAYS_INLINE` below and in particular in `tensor.h` that are still remaining, so as a reader of the code one might really wonder which of the macros to use. I don't think the previous state was pretty (it wasn't), but this just makes it worse by 
(a) employing external macros into core functionality of deal.II
(b) not doing so consistently, but just in some selected places.

I agree this might be a natural step, but we should discuss this in terms of a long-term strategy and the required level of code cleanliness.